src/util/resolveConfig.js

```diff
@@ -1,19 +1,41 @@
-import _ from 'lodash'
+import mergeWith from 'lodash/mergeWith'
+import isFunction from 'lodash/isFunction'
+import defaults from 'lodash/defaults'
+import map from 'lodash/map'
 
 function resolveFunctionKeys(object) {
   return Object.keys(object).reduce((resolved, key) => {
     return {
       ...resolved,
-      [key]: _.isFunction(object[key]) ? object[key](object) : object[key],
+      [key]: isFunction(object[key]) ? object[key](object) : object[key],
     }
   }, {})
 }
 
+function without(object, key) {
+  /* eslint-disable no-unused-vars */
+  return (({ [key]: _, ...rest }) => rest)(object)
+}
+
+function mergeExtensions(theme) {
+  return mergeWith({}, without(theme, 'extend'), theme.extend, (_, extensions, key) => {
```

Comment:
Would not you get the same result spreading it on the argument?

```js
function mergeExtensions({extend, ...theme}) {
  return mergeWith({}, theme, extend, (_, extensions, key) => {
```

Reply:
You're right, so much better, thanks! ❤️